enclave-runtime/src/ocall/attestation_ocall.rs

```diff
@@ -115,6 +111,60 @@ impl EnclaveAttestationOCallApi for OcallApi {
 		Ok((qe_report, quote_vec))
 	}
 
+	fn get_dcap_quote(&self, report: sgx_report_t, quote_size: u32) -> SgxResult<Vec<u8>> {
+		let mut return_quote_buf = vec![0u8; quote_size as usize];
+		let p_quote = return_quote_buf.as_mut_ptr();
+		let p_report = &report as *const sgx_report_t;
+		let mut rt: sgx_status_t = sgx_status_t::SGX_ERROR_UNEXPECTED;
+
+		let result = unsafe {
+			ffi::ocall_get_dcap_quote(&mut rt as *mut sgx_status_t, p_report, p_quote, quote_size)
+		};
+		ensure!(result == sgx_status_t::SGX_SUCCESS, result);
+		ensure!(rt == sgx_status_t::SGX_SUCCESS, rt);
+		let quote_vec: Vec<u8> = Vec::from(&return_quote_buf[..quote_size as usize]);
+		Ok(quote_vec)
+	}
+
+	fn get_qve_report_on_quote(
+		&self,
+		quote: Vec<u8>,
+		current_time: i64,
+		quote_collateral: sgx_ql_qve_collateral_t,
+		qve_report_info: sgx_ql_qe_report_info_t,
+		supplemental_data_size: u32,
+	) -> SgxResult<(u32, sgx_ql_qv_result_t, sgx_ql_qe_report_info_t, Vec<u8>)> {
+		let mut supplemental_data = vec![0u8; supplemental_data_size as usize];
+		let mut qve_report_info_return_value: sgx_ql_qe_report_info_t = qve_report_info;
+		let mut quote_verification_result = sgx_ql_qv_result_t::SGX_QL_QV_RESULT_UNSPECIFIED;
+		let mut collateral_expiration_status = 1u32;
+		let mut rt: sgx_status_t = sgx_status_t::SGX_ERROR_UNEXPECTED;
+
+		let result = unsafe {
+			ffi::ocall_get_qve_report_on_quote(
+				&mut rt as *mut sgx_status_t,
+				quote.as_ptr(),
+				quote.len() as u32,
+				current_time,
+				&quote_collateral as *const sgx_ql_qve_collateral_t,
+				&mut collateral_expiration_status as *mut u32,
+				&mut quote_verification_result as *mut sgx_ql_qv_result_t,
+				&mut qve_report_info_return_value as *mut sgx_ql_qe_report_info_t,
+				supplemental_data.as_mut_ptr(),
+				supplemental_data_size,
+			)
+		};
+		ensure!(result == sgx_status_t::SGX_SUCCESS, result);
+		ensure!(rt == sgx_status_t::SGX_SUCCESS, rt);
+
+		Ok((
+			collateral_expiration_status,
+			quote_verification_result,
+			qve_report_info_return_value,
+			supplemental_data.to_vec(),
+		))
```

Comment:
I think these return types are hard to understand, what would you think about creating a type for this?

```rust
struct QveReport {
  collateral_expiration_status,
 // ... etc
}
```

Reply:
I added the struct you proposed, but I'm not quite sure how far we want to go using it. We could also use it across the FFI boundary. Not sure that we want to do that.